### DISCUSSION TITLE
Download all attachments in a folder/for an item

### DISCUSSION THREAD

**Discussion by cboulanger** [windingwind/zotero-actions-tags#266] [Ideas]
Feature: I would like to have an action for collections that would download all attachments for all items in a collection. 

Background: I try to save space on my laptop by downloading attachments on demand. However, sometimes it is necessary to have the attachments for a particular collection for situations where I have no internet connection.

Has someone tried that already? If not: a few pointers to the relevant API parts would help, then I can write it myself.

**Comment by windingwind** [comment]
Havn't tried, but I think this might work:

```js
if (!item) return;

let attachments = [];
if (item.isAttachment()) {
  attachments = [item];
} else if (item.isRegularItem()) {
  attachments = Zotero.Items.get(item.getAttachments());
} else {
  return;
}

for (let attachment of attachments) {
  let path = await attachment.getFilePathAsync();
  if (!path) {
    let fileSyncingEnabled = Zotero.Sync.Storage.Local.getEnabledForLibrary(
      item.libraryID
    );
    if (!fileSyncingEnabled) {
      return "File sync is disabled";
    }

    try {
      let results = await Zotero.Sync.Runner.downloadFile(attachment);
      if (!results || !results.localChanges) {
        return "Download failed";
      }
    } catch (e) {
      Zotero.debug(e);
    }
  }
}
```

**Comment by cboulanger** [comment]
Thanks! I updated it a bit, works like a charm: https://github.com/windingwind/zotero-actions-tags/discussions/267
